backend/apps/ecommerce/mutations.py

```diff
@@ -0,0 +1,154 @@
+                    f"{order.order_id}-{order.payment_attempt}"
+                )
+                # If order canceled: retry, if reserved: stop attempt and restore quantity
+                if status_success and status == "RESERVE":
+                    order.payment_status = Order.PaymentStatus.RESERVED
+                    order.save()
+                    order.product.restore_quantity(order)
+                    return InitiateOrder(order_id=order.order_id)
+
+                # Retry
+                order.payment_attempt = order.payment_attempt + 1
+                order.payment_status = Order.PaymentStatus.INITIATED
+                order.save()
+
+        except Order.DoesNotExist:
+            org_name = product.organization.slug[:13].replace(" ", "-")
+
+            order_id = f"{org_name}-{uuid.uuid4().hex}"
+
+            order = Order()
+            order.order_id = order_id
+            order.product = product
+            order.user = user
+            order.quantity = quantity
+            order.total_price = product.price * quantity
+
+            order.save()
+
+        redirect = InitiateOrder.vipps_api.initiate_payment(order)
+
+        return InitiateOrder(redirect=redirect)
+
+
+class AttemptCapturePayment(graphene.Mutation):
+    # Polling request to capture payment in case callback does not succeed
+    # Also returns payment status
+
+    status = graphene.String()
+    order = graphene.Field(OrderType)
+    vipps_api = VippsApi()
+
+    class Arguments:
+        order_id = graphene.ID(required=True)
+
+    @login_required
+    def mutate(self, info, order_id):
+        with transaction.atomic():
+            try:
+                # Acquire DB lock for the order (no other process can change it)
+                order = Order.objects.select_for_update().get(pk=order_id)
+            except Order.DoesNotExist:
+                raise ValueError("Ugyldig ordre")
+
+            if order.user != info.context.user:
+                raise PermissionDenied("Du har ikke tilgang til denne ordren")
+
+            if order.payment_status == Order.PaymentStatus.INITIATED:
+                # Update status according to Vipps payment details
+                status, status_success = AttemptCapturePayment.vipps_api.get_payment_status(
+                    f"{order.order_id}-{order.payment_attempt}"
+                )
+
+                if status_success and status == "RESERVE":
+                    order.payment_status = Order.PaymentStatus.RESERVED
+                    order.save()
+                elif status_success and status == "CANCEL":
+                    order.payment_status = Order.PaymentStatus.CANCELLED
+                    order.save()
+                    # Order went from initiated to cancelled, restore quantity
+                    order.product.restore_quantity(order)
+
+            if order.payment_status == Order.PaymentStatus.RESERVED:
```

Comment:
```suggestion
            elif order.payment_status == Order.PaymentStatus.RESERVED:
```

Reply:
This suggestion is wrong as we want to capture the payment here if it was changed from `INITIATED` to `RESERVED` in the previous if-check.